cmd/sdkstats/main.go

```diff
@@ -7,46 +7,141 @@ import (
+			},
+			VersionParser: &gomodVersionParser{
+				searchQ: "replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v",
+			},
+		},
+		sdkstats.RepoMetadataQuery{
+			ProjectType:   "gopkg.toml",
+			Queries:       []string{"filename:Gopkg.toml github.com/operator-framework/operator-sdk"},
+			VersionParser: &unknownVersionParser{},
+		},
+	}
+	// GetStats function for Query String from 'queries', These Strings are specific to Operator-SDK patterns.
+	collectStats := [][]sdkstats.RepoMetadata{}
 
 	for _, r := range queries {
-		stats, err := sdkstats.GetStats(client, r)
-		if len(stats) == 0 {
-			log.Fatal("GITHUB API is down at time, Please try after few minutes")
-		} else {
-			fileName := stats[0].ProjectType + ".json"
-			if err != nil {
-				fmt.Println(err)
+		stats, err := sdkstats.GetStats(ctx, tc, r)
+		if err != nil {
+			fmt.Printf("Failed to get stats for query  %q %v\n", r, err)
+		}
+		collectStats = append(collectStats, stats)
+	}
+	fileName := "Search_Results.json"
+	file, _ := json.MarshalIndent(collectStats, "", " ")
+	_ = ioutil.WriteFile(fileName, file, 0644)
+	fmt.Println("Results are written in Search_Results.json")
+}
+
+// Parse the given Code result to search Text Matches for Version number.
+func (p baseVersionParser) ParseVersion(codeResults github.CodeResult) (string, error) {
+	var version string
+	var s, v []string
+
+	for _, r := range codeResults.TextMatches {
+		if strings.Contains(r.GetFragment(), p.searchQ) {
+			value := r.GetFragment()
+			s = strings.Split(value, "\n")
+		stLoop:
+			for _, st := range s {
+				if strings.Contains(st, p.searchQ) {
+					v = strings.Split(st, ":")
+					if len(v) == 0 {
+						version = "N/A"
+					} else {
+						version = v[1]
+					}
+					break stLoop
+				}
 			}
-			fmt.Println("Total count: ", len(stats))
-			file, _ := json.MarshalIndent(stats, "", " ")
-			_ = ioutil.WriteFile(fileName, file, 0644)
-			fmt.Println("Results are written in ", fileName)
+		} else if strings.Contains(r.GetFragment(), p.searchLatest) {
+			version = "v0.11.0"
 		}
 	}
+	return version, nil
+}
+
+// Parse the given Code result to search Text Matches for Version number.
+func (p gomodVersionParser) ParseVersion(codeResults github.CodeResult) (string, error) {
+	var version string
+	var s, v []string
+
+	for _, r := range codeResults.TextMatches {
+		if strings.Contains(r.GetFragment(), p.searchQ) {
+			value := r.GetFragment()
+			s = strings.Split(value, "\n")
+		stLoop:
+			for _, st := range s {
+				if strings.Contains(st, p.searchQ) {
+					v = strings.Split(st, " v")
+					if len(v) == 0 {
+						version = "N/A"
+					} else {
+						version = v[1]
+					}
+					break stLoop
+				}
+			}
+		}
+	}
+	return version, nil
+}
 
+// Parse the given Code result to search Text Matches for Version number.
+func (p unknownVersionParser) ParseVersion(codeResults github.CodeResult) (string, error) {
+	version := "N/A"
```

Comment:
Nit: This isn't really `"N/A"` so much as `"unknown"` right? `Gopkg.toml` files DO have the SDK version in them, but we're just not able to parse it.

Also, there's no need to declare a variable and then return it. It's probably better to do `return "unknown", nil`.

Reply:
done